docs/arch/adr-0001-style-guide.md

```diff
@@ -0,0 +1,76 @@
+# Style Guide
+
+
+## Context
+
+We need a style guide for our actions repo. This structure provides the following benefits:
+- developers can spend more time on solving the problem
+- PR reviews are limited to discussing intent
+- format related feedback can point to a collective style guide instead of personal preferences
+- consumers know how to find what they're looking for
+- developers know where to look when fixing something
+
+## Options
+
+This style guide is a collection of individual options. Instead of providing permutations of these options,
+this ADR provides a section for an initial proposal for several style guide topics and a section
+that contains the agreed upon formats. This ADR should be updated to remove reference to the initial
+proposal prior to merging in order to avoid confusion.
+
+As an alternative, each topic could be its own ADR. Reviewers may request this approach, which will require
+breaking this ADR into several ADRs and associated PRs.
+
+## Proposal
+
+### Actions
+- actions live at the repo root so that usage looks like `uses: dbt-labs/actions/<action-name>@<tag>`
+- actions live in directories with the format `<platform/tool>/<resource>/<action>`, e.g.
+    - `./hatch/environment/create`
+    - `./pypi/release/view`
+- actions have a generic scope with a focused context to support reusability
+- actions perform a single task on a single object within the context of a platform/tool
+
+### Workflows
+- workflows need to live in `.github/workflows`
+- we prefer actions to workflows
```

Comment:
I don't agree with this.  I want to avoid composition of the same 7 actions to accomplish one thing. Having to dig multiple levels down is also frustrating and makes it unclear what's happening.

Reply:
I struggle with building new workflows off of other workflows for a few reasons:

- reusable workflows are forced to be their own jobs in the calling workflow
  - I can't embed them in a job
  - I can't pass environment variables to workflows, which is especially a problem when those change by repo (e.g. connection credentials)
  - I need to create a bunch of inputs to pass to them, to get a bunch of outputs from them, to pass those outputs to another job, in many cases to perform one or two additional steps
- any reusable workflow that needs access to the code requires adding repo and branch/ref as arguments
  - with an action, you do the checkout in the calling workflow and the called action can focus on the task
- reusable workflows tend to take on too much scope, making them difficult to combine
  - for example, we currently run tests twice during our release for each adapter because both the bumpversion/changelog workflow and build/test workflow runs tests; ideally there is an action for bumpversion, an action for changelog generation, an action for build, and these can all be selected al a carte as needed
  - stated another way, actions should be created to perform a specific task, which should minimize maintenance for a particular action, similar to how python functions should be built